eap_backend/eap_api/views.py

```diff
@@ -97,6 +93,42 @@ class DetailAssuranceView(generics.RetrieveUpdateDestroyAPIView):
     serializer_class = AssuranceCaseSerializer
 
 
+def get_case_id(item):
+    """Return the id of the case in which this item is. Works for all item types."""
+    # In some cases, when there's a ManyToManyField, instead of the parent item, we get
+    # an iterable that can potentially list all the parents. In that case, just pick the
+    # first.
+    if hasattr(item, "first"):
+        item = item.first()
+    if isinstance(item, models.AssuranceCase):
+        return item.id
+    elif hasattr(item, "assurance_case") and item.assurance_case is not None:
```

Comment:
This is fine as it is, but I'm wondering if we could use the TYPE_DICT to get the parent type of each item rather than all the "if/elif?

Reply:
Done